### PR TITLE
Covering all tutorials from step-5 to step-20 by CTest.

### PR DESCRIPTION
**Pull request by vyushut**
test/examples should include an .output file and a .diff file for each tutorial from examples/step-*/step-*.cc. The diff file typically reduces the refinement level and suppresses the non-textual output the CTest runs.

### CODE REVIEW

File: tests/examples/step-5.output [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+Cycle 3:
+   Number of active cells: 1280
+   Total number of cells: 1700
+   Number of degrees of freedom: 1313
+   38 CG iterations needed to obtain convergence.
+Cycle 4:
+   Number of active cells: 5120
+   Total number of cells: 6820
+   Number of degrees of freedom: 5185
+   70 CG iterations needed to obtain convergence.
```
Comment (tjhei) [inline tests/examples/step-5.output:25]: we might need to do something like https://github.com/dealii/dealii/blob/master/tests/examples/step-12.diff to make it robust to small changes to iteration numbers. Let's see what the tester says.

**Review by tjhei — APPROVED** [review]
Thank you.

**Review by kronbichler — APPROVED** [review]

**Comment by kronbichler** [comment]
As anticipated by @tjhei we have some issue with iteration counts at least in step-18 (from the Jenkins MPI tester):
```
examples/step-18.mpirun=2.debug: DIFF failed. ------ First 20 lines of numdiff output:
----------------
##30      #:4   <== 126
##30      #:4   ==> 127
```

**Review by tjhei — APPROVED** [review]

**Comment by tjhei** [comment]
```
170/183 Test  #11: examples/step-18.mpirun=2.debug ...............................................***Timeout 600.11 sec```

**Comment by tjhei** [comment]
Not quite:
```
examples/step-18.mpirun=2.debug: DIFF failed. ------ First 20 lines of numdiff output:
11: ----------------
11: ##18      <== Timestep 2 at time 2
11:           ==>
```
:-)

**Pull request merged**